### DISCUSSION TITLE
Static Portable Build.

### DISCUSSION THREAD

**Discussion by Alex313031** [geany/geany#3742] [General]
Hello, I have a compiler optimized fork that I build for Linux and Windows. > https://github.com/Alex313031/geany-ng On Windows one can make a "portable" version, by just zipping up the artifacts in the installer.

However, on Linux, just like alot of programs that use make install, this doesn't work. It expects libraries to be in certain places. Simply placing all the artifacts in one dir doesn't allow it to run.

Is there a way one could make a "Static" build, or perhaps adjust the prefix so that one could make a "portable" build.

**Comment by elextr** [comment]
> adjust the prefix so that one could make a "portable" build.

The usual `./configure --prefix=/foo/blah` or as `README` says `meson configure --prefix=/foo/blah`

**Reply by b4n** [reply]
Does that actually solve anything if there isn't */foo/blah* on the target system?

**Reply by Alex313031** [reply]
It worked. With this and the below comment about --enable-binreloc.
I was also able to make a Windows release with MSYS2.
Thanks yall

**Comment by b4n** [comment]
Look up the relocatable option (`--enable-binreloc` or something), it should work. If you need extra libraries packed, you should be able to ship them next to *libgeany.so*.

**Reply by Alex313031** [reply]
It worked. Yeah I included some GTK libs.
I was also able to make a Windows release with MSYS2.
Thanks yall
